pkg/aws/client/client.go

```diff
@@ -443,3 +262,152 @@ func (c *Client) DeleteBucketIfExists(ctx context.Context, bucket string) error
 	}
 	return nil
 }
+
+// The following functions are only temporary needed due to https://github.com/gardener/gardener/issues/129.
+
+// ListKubernetesELBs returns the list of ELB loadbalancers in the given <vpcID> tagged with <clusterName>.
+func (c *Client) ListKubernetesELBs(ctx context.Context, vpcID, clusterName string) ([]string, error) {
+	var (
+		loadBalancerNamesInVPC      []*string
+		loadBalancerNamesForCluster []string
+	)
+
+	if err := c.ELB.DescribeLoadBalancersPagesWithContext(ctx, &elb.DescribeLoadBalancersInput{}, func(page *elb.DescribeLoadBalancersOutput, lastPage bool) bool {
+		for _, lb := range page.LoadBalancerDescriptions {
+			if lb.VPCId != nil && *lb.VPCId == vpcID {
+				loadBalancerNamesInVPC = append(loadBalancerNamesInVPC, lb.LoadBalancerName)
+			}
+		}
+		return !lastPage
+	}); err != nil {
+		return nil, err
+	}
+
+	tags, err := c.ELB.DescribeTagsWithContext(ctx, &elb.DescribeTagsInput{LoadBalancerNames: loadBalancerNamesInVPC})
```

Comment:
In the integration test and locally this step fails with:

```
{"level":"error","ts":"2021-03-18T09:53:19.707+0200","logger":"controller-runtime.manager.controller.infrastructure_controller","msg":"Reconciler error","name":"aws-local","namespace":"shoot--foo--aws-local","error":"1 error occurred:\n\t* Failed to destroy load balancers and security groups: InvalidParameter: 1 validation error(s) found.\n- missing required field, DescribeTagsInput.LoadBalancerNames.\n\n\n","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/github.com/gardener/gardener-extension-provider-aws/vendor/github.com/go-logr/zapr/zapr.go:132\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/src/github.com/gardener/gardener-extension-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:267\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/gardener/gardener-extension-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:235\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.1\n\t/go/src/github.com/gardener/gardener-extension-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:198\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\t/go/src/github.com/gardener/gardener-extension-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/go/src/github.com/gardener/gardener-extension-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/go/src/github.com/gardener/gardener-extension-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/gardener/gardener-extension-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\t/go/src/github.com/gardener/gardener-extension-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\n\t/go/src/github.com/gardener/gardener-extension-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:99"}
```

So I guess the reason for this is that `loadBalancerNamesInVPC` could be empty slice.

Reply:
Thanks @ialidzhikov, I added an early exit: 676bef4